File: Java/novenoEjercicio.java
import java.util.Random;
import java.util.Scanner;

public class novenoEjercicio {
    public static void main(String[] args) {
        int intentos = 5;
        int[] numeros = new int[intentos];
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();
        int[] aleatorio = new int[1];

        for (int i = 0; i < aleatorio.length; i++) {
            aleatorio[i] = random.nextInt(101);
        }
        System.out.println("Tienes 5 intentos para adivinar el número aleatorio");
        for (int i = 0; i < intentos; i++) {
            System.out.println("digite un numero");
            int num = esEntero(scanner);
            numeros[i]=num;
            buscarNumero(aleatorio, num);
        }
        if (intentos==0){
            System.out.println("¡lo siento, pero no adivinaste!");
        }
        System.out.println("El número a adivinar era: ");
        for (int aleator: aleatorio){
            System.out.println(aleator);
        }
        System.out.print("Los números que digitaste: ");
        for (int numero: numeros){
            System.out.print(numero + " ");
        }
    }
    public static boolean buscarNumero(int[] vector,int numero) {
        for (int num : vector) {
            if (num > numero) {
                System.out.println("¡Frio, frio!");
                return true;
            }else if (num<numero){
                System.out.println("¡Caliente, Caliente!");
            }else{
                System.out.println("Haz atinado");
            }

        }
        return false;
    }
    public static int esEntero(Scanner scanner){
        while (true){
            try {
                return Integer.parseInt(scanner.next());
            }catch (NumberFormatException e){
                System.out.println("Error: ingrese un nuevo entero");
            }
        }
    }
}

File: Python/noveno_ejercicio.py
from utils import es_numero_entero
from rich.console import Console
console = Console()
import random
def buscar_numero(numero, vector):
    for num in vector:
        if num > numero:
            console.print("[italic blue]¡Frio, frio![italic blue]")
            return True
        elif num < numero:
            console.print("[italic red]¡Caliente, Caliente![italic red]")
            return
        else:
            console.print("[italic green]¡Haz atinado![italic green]")
    console.print("[bold red]El número no está en el vector.[bold red]")

def main(intentos=5):
    i:0
    numeros=[0]*intentos
    aleatorio=[random.randint(0,100) for i in range(1)]
    console.print("[bold magenta]Tienes 5 intentos para adivinar el número aleatorio[bold magenta]")
    for i in range(intentos):
        num = es_numero_entero(console.input("[bold magenta]digite un numero\n[bold magenta]"))
        numeros[i] = num
        buscar_numero(num, aleatorio)
    if intentos==0:
        console.print("[italic red]¡lo siento, pero no adivinaste![italic red]")
    console.print("[italic green]El número a adivinar era: [italic green]")
    console.print(f"[italic blue]{aleatorio}[italic blue]")
    console.print("[bold yellow]Los números que digitaste: [bold yellow]")
    for numero in numeros:
        console.print(f"{numero}")

if __name__=="__main__":
    main()